ykman/cli/piv.py

```diff
@@ -648,26 +659,41 @@ def change_management_key(
     # If an old stored key needs to be cleared, the PIN is needed.
     if not protect and controller.has_stored_key:
         if pin:
-            _verify_pin(ctx, controller, pin)
+            _verify_pin(ctx, controller, pin, no_prompt=force)
         else:
             force or click.confirm(
                     'The current management key is stored on the YubiKey'
                     ' and will not be cleared if no PIN is provided. Continue?',
                     abort=True)
 
     if not new_management_key and not protect:
-        if not force:
+        if generate:
+            new_management_key = generate_random_management_key()
+
+            if not protect:
+                click.echo(
+                    'Generated management key: {}'.format(
+                        b2a_hex(new_management_key).decode('utf-8')))
+
+        elif force:
+            ctx.fail('New management key not given. Please remove the --force '
+                     'flag, or set the --generate flag or the '
+                     '--new-management-key option.')
+
+        else:
```

Comment:
So for the static password command we actually dropped the "blank to randomly generate" functionality, I'd suggest we do the same thing here to simplify the code and have similar behaviour.

Reply:
Sounds good to me!